src/qat/purr/backends/qiskit_simulator.py

```diff
@@ -247,14 +281,40 @@ def execute(self, builder: QiskitBuilder) -> Dict[str, Any]:
             AerSimulator._DEFAULT_CONFIGURATION | {"open_pulse": False}
         )
         aer_config.n_qubits = self.model.qubit_count
-        qasm_sim = AerSimulator(aer_config, noise_model=builder.model.noise_model)
+
+        # AerSimulator can call a range of backends. We allow these to be choosen, along with
+        # any relevant arguments.
+        config = self.create_config(kwargs)
+        qasm_sim = AerSimulator(aer_config, noise_model=builder.model.noise_model, **config)
 
         try:
             job = qasm_sim.run(
                 transpile(circuit, qasm_sim, coupling_map=coupling_map),
                 shots=builder.shot_count,
             )
             results = job.result()
+
+            # The job can fail if it runs into e.g. memory issues using the statevector
+            # simulator. In this case, and if no method is provided, try again using an
+            # MPS backend.
```

Comment:
the optimistic retry could also be configured as a sequence in qat config .... The tail of which is the default or the fail back route